purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields
+from odoo.tests import Form, TransactionCase, tagged
+
+
+@tagged("-at_install", "post_install")
+class TestPurchaseOrderSecondaryUnit(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Active multiple units of measure security group for user
+        cls.env.user.groups_id = [(4, cls.env.ref("uom.group_uom").id)]
+        cls.product_uom_kg = cls.env.ref("uom.product_uom_kgm")
+        cls.product_uom_gram = cls.env.ref("uom.product_uom_gram")
+        cls.product_uom_unit = cls.env.ref("uom.product_uom_unit")
+        # Create a product
+        product_form = Form(cls.env["product.product"])
+        product_form.name = "Test"
+        product_form.uom_id = cls.product_uom_kg
+        product_form.uom_po_id = cls.product_uom_kg
+        cls.product = product_form.save()
+        # Set a secondary unit on the template of the previously created product
+        with Form(cls.product.product_tmpl_id) as template_form:
+            with template_form.secondary_uom_ids.new() as secondary_uom_form:
+                secondary_uom_form.name = "unit-700"
+                secondary_uom_form.uom_id = cls.product_uom_unit
+                secondary_uom_form.factor = 0.7
+        cls.secondary_unit = cls.product.product_tmpl_id.secondary_uom_ids
+        cls.product.purchase_secondary_uom_id = cls.secondary_unit.id
+        cls.partner = cls.env["res.partner"].create({"name": "test - partner"})
+        cls.purchase_order_obj = cls.env["purchase.order"]
+        po_val = {
+            "partner_id": cls.partner.id,
+            "company_id": cls.env.company.id,
+            "order_line": [
+                (
+                    0,
+                    0,
+                    {
+                        "name": cls.product.name,
+                        "product_id": cls.product.id,
+                        "product_qty": 1,
+                        "product_uom": cls.product.uom_id.id,
+                        "price_unit": 1000.00,
+                        "date_planned": fields.Datetime.now(),
+                    },
+                )
```

Comment:
Could you use the [Command](https://github.com/odoo/odoo/blob/370de95190cf07f17281f0ed6365b6806663cac2/odoo/fields.py#L4227-L4256) class? It is more readable. And check the remaining code, please.

```suggestion
            "order_line": [
                Command.create(
                    {
                        "name": cls.product.name,
                        "product_id": cls.product.id,
                        "product_qty": 1,
                        "product_uom": cls.product.uom_id.id,
                        "price_unit": 1000.00,
                        "date_planned": fields.Datetime.now(),
                    },
                )
```

Reply:
Done